### PR TITLE
Add i18n for screen reader section labels

### LINKED ISSUE #1586: Make screen reader section labels translatable
We use custom configuration `rehype-autolink-headings` to wrap our headings and add the clickable link icon. This includes a screenreader label which we inject but this is currently hardcoded in English:

https://github.com/withastro/docs/blob/41ea6653d78cfa90d96cd41019dbf1440a07f2af/astro.config.ts#L37

We should make this translatable like our other language strings.

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- Translated content
- Changes to the docs site code

#### Description

- Closes #1586 

This PR adds a new rehype plugin to translate the screen reader anchor labels injected by `rehype-autolink-headings`. If the current language doesn't have a translation yet, it will fallback to English.

| PT-BR (translated)  | ES (fallback) |
| ------------- | ------------- |
| image | image |

### CODE REVIEW

File: src/i18n/util.ts [hunk 1]
```diff
@@ -104,3 +104,13 @@ export function useTranslations(
 		return str;
 	};
 }
+
+export function useTranslationsFromString(
```
Comment (delucis) [inline src/i18n/util.ts:108]: I might suggest renaming this `useTranslationsForLang`. What do you think?

File: src/i18n/util.ts [hunk 2]
```diff
@@ -104,3 +104,13 @@ export function useTranslations(
 		return str;
```
Comment (delucis) [inline src/i18n/util.ts:104]: Could we refactor `useTranslations` to use the new function and avoid duplication?

```js
export function useTranslations(
	Astro: Readonly<AstroGlobal>
): (key: UIDictionaryKeys) => string | undefined {
	const lang = getLanguageFromURL(Astro.url.pathname) || 'en';
	return useTranslationsForLang(lang);
}
```

**Review by delucis — COMMENTED** [review]
Clever solution! Shame `rehype-autolink-headings` doesn’t provide `file` etc. as context to its different methods so we could just plug this in there. Made a small naming suggestion & refactor recommendation.

LOVE that this is your first PR back — i18n + a11y in one go 🙌

File: src/i18n/util.ts (hunk 2 above)
Comment (Yan-Thomas) [reply src/i18n/util.ts:104]: Agree, smart suggestions!

**Review by delucis — APPROVED** [review]
LGTM! Welcome back @Yan-Thomas 🚀

**Pull request merged**